### ISSUE TITLE
Build a windows installer?

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#122]
installing python and dependencies causes a lot of support calls. While pip based installs are better for python users, non python users might benefit from a simpler installation method.

perhaps nuitka can help?  other ideas?

**Comment by petersilva** [comment]
Stuff @iMacadan discovered:

* if you install in a from a git cloned, using pip install, it creates sr_entry.exe, (where entry=subscribe,shovel, etc...) sr_entry-script.py and places them in the same directory as python.exe.
*If you install from pypi.python.org, only the sr_entry.exe is created with no -script.py´s.

In both cases #118 applies...

**Comment by petersilva** [comment]
We do not want to build an installer, and are trying to avoid it, but this thread is a place holder for the drivers that are pushing us that way.

**Comment by iMacadan** [comment]
About our discussion about finding a Windows deamon-like solution, can this help ?
https://daemoniker.readthedocs.io/en/latest/index.html

**Comment by benlapETS** [comment]
IMHO I would say that having a msi or .exe installer to deploy on Windows is the basic standard and the long standing way to do it. Because Windows user are used with that practice that is visual and simple. I stand by it, even if the size of the dependencies (which Peter estimate to 300MB) is quite bigger than sarracenia itself and no matters that the operations required to install it may be simpler (ie 1. install python, install pip, pip install metpx-sarracenia, ...).
In short, I think it would please our Windows users. I am still not sure of what tool would best suite this need, but with a basic google search, i found this : https://cyrille.rossant.net/create-a-standalone-windows-installer-for-your-python-application/

**Comment by petersilva** [comment]
Did you read the top comment in your article? It complains about a 600MB package ... but I agree that it is worth it to make users' lives easier. It would be worth a try... go ahead and try to build one... we'll see how it goes.

**Comment by benlapETS** [comment]
Ok, thx, I will explore this way better today as you agree to give it a try, like I said it was a basic search I didn't read everything and I notice today that this article is a bit old. Worst of all, Ive red a comment today that say this :
"Tried doing this to create an installer for my 5kb python script. Bundling it with Python caused the folder to be more than 1.2gb, which I guess is unacceptable for most users"
Seems to me that there might be a better way to do it, I will dig more for other options.

**Comment by benlapETS** [comment]
I've completed a script that generate a Windows installer using pynsist. This is a first untested, undocumented yet implementation that works on my side when I install on Windows 10.

**Comment by benlapETS** [comment]
Just remove the txt extension (I promise this is not a virus) lolll
[Sarracenia_v2.19.01b1.exe.txt](https://github.com/MetPX/sarracenia/files/2788352/Sarracenia_v2.19.01b1.exe.txt)

**Comment by petersilva** [comment]
prototype is here: https://hpfx.collab.science.gc.ca/~pas037/Sarracenia_Releases/
Successfully installed on two windows 10 desktop machines. it seems to work ok, just need some long running tests on internal machines.
